src/domain/actuator.py

```diff
@@ -0,0 +1,34 @@
+"""Actuator module"""
+
+import logging
+from abc import abstractmethod
+from enum import Enum
+
+logger = logging.getLogger(__name__)
+
+# List of Actuators
+actuators = []
+
+
+class Actuator:
+    """Base class of an actuator."""
+
+    class ActuatorTypes(Enum):
+        Semaphore = "Semaphore"
+        Feeder = "Feeder"
+
+    def __init__(self, id, enabled=False):
+        self.type = None
+        self.id = id
+        self.enabled = enabled
+        self.stop_thread = False
+
+    @abstractmethod
+    def serialize(self):
+        """Method to serialize Actuator object into file."""
+        pass
```

Comment:
pass is redundant when you've a docstring. But you can keep it here for more clarity (albeit the abstractmethod decorator gives already a good idea of what this method is).